### PR TITLE
[SYCLomatic]Support the migration of l2CacheSize in cudaDeviceProp

### PR DESCRIPTION
**Pull request by wenhuiNi**
Signed-off-by: Ni, Wenhui <wenhui.ni@intel.com>

### CODE REVIEW

**Review by yihwang-nv — APPROVED** [review]
LGTM

**Review by ziranzha — APPROVED** [review]

File: clang/runtime/dpct-rt/include/device.hpp [hunk 1]
```diff
@@ -136,6 +136,9 @@ class device_info {
   unsigned int get_memory_bus_width() const { return _memory_bus_width; }
   uint32_t get_device_id() const { return _device_id; }
   std::array<unsigned char, 16> get_uuid() const { return _uuid; }
+  /// Returns estimated size of L2 cache in bytes.
+  unsigned int get_L2_cache_size() { return _L2_cache_size; }
```
Comment (tangjj11) [inline clang/runtime/dpct-rt/include/device.hpp]:
```suggestion
  unsigned int get_L2_cache_size() const { return _L2_cache_size; }
```

**Review by tangjj11 — APPROVED** [review]

File: clang/runtime/dpct-rt/include/device.hpp [hunk 2]
```diff
@@ -393,6 +401,10 @@ Use 64 bits as memory_bus_width default value."
     // according to device properties.
     prop.set_max_register_size_per_work_group(65536);
 
+    // Set estimates max L2 cache size(16MB), feel free to update the value
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/device.hpp]: please check if we can use info::device::global_mem_cache_ size

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**